apps/web/components/DestinationCalendarSelector.tsx

```diff
@@ -30,13 +31,24 @@ const DestinationCalendarSelector = ({
         .map((connected) => connected.calendars ?? [])
         .flat()
         .find((cal) => cal.externalId === value);
-
       if (selected) {
         setSelectedOption({
           value: `${selected.integration}:${selected.externalId}`,
           label: selected.name || "",
         });
       }
+      else {
+        let defaultValue = query.data?.connectedCalendars
+          .map((connected) => connected.calendars ?? []);
+        if (defaultValue?.length > 0) {
+          defaultValue = defaultValue[0];
+          setSelectedOption({
+            value: `${defaultValue[0].integration}:${defaultValue[0].externalId}`,
+            label: defaultValue[0].name || "",
+          });
+        }
+
+      }
```

Comment:
Let's try to keep this PR with specific code for the issue mentioned. Otherwise it will be harder to revert if needed.